cli/src/commands/git/push.rs

```diff
@@ -966,25 +972,35 @@ fn find_bookmarks_targeted_by_revisions<'a>(
 ) -> Result<Vec<(&'a RefName, LocalAndRemoteRef<'a>)>, CommandError> {
     let mut revision_commit_ids = HashSet::new();
     if use_default_revset {
-        // remote_bookmarks(remote=<remote>)..@
-        let workspace_name = workspace_command.workspace_name();
-        let expression = RevsetExpression::remote_bookmarks(
-            StringPattern::everything(),
-            StringPattern::exact(remote),
-            None,
-        )
-        .range(&RevsetExpression::working_copy(workspace_name.to_owned()))
-        .intersection(&RevsetExpression::bookmarks(StringPattern::everything()));
+        // Get the default revset specified by the user.
+        let push_revset = workspace_command
+            .settings()
+            .get_string("revsets.git-push")?;
+        let mut context = workspace_command.env().revset_parse_context();
+        context.local_variables.insert(
+            "remote",
+            ExpressionNode {
+                kind: ExpressionKind::StringPattern {
+                    kind: "exact",
+                    value: remote.as_str().to_string(),
+                },
```

Comment:
Can you add test that would fail if this were substring pattern?

Reply:
I tried to add one. PTAL